agent/tags_test.go

```diff
@@ -82,6 +82,32 @@ func TestFetchingTagsFromEC2Tags(t *testing.T) {
 		[]string{"custom_tag=true"})
 }
 
+func TestFetchingTagsFromECS(t *testing.T) {
+	fetcher := &tagFetcher{
+		ecsMetaDataDefault: func() (map[string]string, error) {
+			return map[string]string{
+				`ecs:container-name`: "ecs-buildkite-agent-blahblah",
+				`ecs:image`:          "buildkite/agent",
```

Comment:
super nit here:
```suggestion
				"ecs:container-name": "ecs-buildkite-agent-blahblah",
				"ecs:image":          "buildkite/agent",
```

i realise that there's a bunch of prior art in the repo of using backtick strings for hash keys(? i guess? seems like a weird choice from my forebears), but it's best to avoid them unless we actually need them in future, i think.